packages/web/src/js/FileUploader.ts

```diff
@@ -326,23 +340,62 @@ class FileUploader extends BaseComponent {
     this.dragReset();
   }
 
-  addToQueue(file: File) {
+  addToQueue(
+    file: File,
+    meta?: FileQueueValueMetaType,
+    callback?: (key: string, file: File, meta?: FileQueueValueMetaType) => void,
+  ) {
     try {
       EventHandler.trigger(this.wrapper, EVENT_QUEUE_FILE, { fileQueue: this.fileQueue, currentFile: file });
       this.checkAllowedFileType(file);
       this.checkAllowedFileSize(file);
       this.checkFileQueueDuplicity(file);
       this.checkQueueLimit();
-      this.appendToList(file);
+      this.appendToList(file, meta);
       this.updateDropZoneVisibility();
       this.updateNameAttribute();
+      callback && callback(this.getUpdatedFileName(file.name), file, meta);
       EventHandler.trigger(this.wrapper, EVENT_QUEUED_FILE, { fileQueue: this.fileQueue, currentFile: file });
     } catch (error) {
       EventHandler.trigger(this.wrapper, EVENT_ERROR, { validationText: error.message });
     }
   }
 
-  removeFromQueue(name: string) {
+  updateQueue(
+    name: string,
+    file: File,
+    meta?: FileQueueValueMetaType,
+    callback?: (key: string, file: File, meta?: FileQueueValueMetaType) => void,
+  ) {
+    if (this.fileQueue.has(name)) {
+      const newValue: FileQueueValueType = { file };
+      if (meta) {
+        newValue.meta = meta;
+      }
+
+      this.fileQueue.set(name, newValue);
+
+      const itemImgElement = SelectorEngine.findOne(`#${name} .FileUploaderAttachment__image img`);
+      if (
```

Comment:
`['x', 'y', 'width', 'height'].every((coord) => meta[coord] != null)` => maybe `function isCoordsInMeta(meta)`

Reply:
created new method and fixed the condition